include/lego_sensor_class.h

```diff
@@ -124,6 +125,7 @@ struct lego_sensor_device {
 	ssize_t (*direct_write)(void *context, char *data, loff_t off, size_t count);
 	int (* get_poll_ms)(void *context);
 	int (* set_poll_ms)(void *context, unsigned value);
+	ssize_t (*text_value_read)(char *data, size_t count);
```

Comment:
And no count anymore - I just return a pointer, and if it's not NULL or ERR (there's a macro for that) I do an `snprintf()` otherwise return 0